src/stan/lang/ast/expr_type_def.hpp

```diff
@@ -1,15 +1,23 @@
 #ifndef STAN_LANG_AST_EXPR_TYPE_DEF_HPP
 #define STAN_LANG_AST_EXPR_TYPE_DEF_HPP
 
+#include <stan/lang/ast/ill_formed_type.hpp>
+#include <stan/lang/ast/void_type.hpp>
+#include <stan/lang/ast/int_type.hpp>
+#include <stan/lang/ast/double_type.hpp>
+#include <stan/lang/ast/vector_type.hpp>
+#include <stan/lang/ast/row_vector_type.hpp>
+#include <stan/lang/ast/matrix_type.hpp>
 #include <stan/lang/ast/base_expr_type.hpp>
+
 #include <stan/lang/ast/expr_type.hpp>
 #include <stan/lang/ast/fun/write_base_expr_type.hpp>
 #include <ostream>
 
 namespace stan {
   namespace lang {
 
-    expr_type::expr_type() : base_type_(ILL_FORMED_T), num_dims_(0) { }
+    expr_type::expr_type() : base_type_(ill_formed_type()), num_dims_(0) { }
 
     expr_type::expr_type(const base_expr_type base_type)
```

Comment:
Any argument that's not primitive should be by constant reference.

[required change]

Reply:
fixed.